src/plugins/discover/public/utils/get_field_type_name.ts

```diff
@@ -7,70 +7,89 @@
+
+export function getFieldTypeName(type?: string) {
   switch (type) {
-    case 'boolean':
+    case KNOWN_FIELD_TYPES.BOOLEAN:
       return i18n.translate('discover.fieldNameIcons.booleanAriaLabel', {
         defaultMessage: 'Boolean field',
       });
-    case 'conflict':
+    case KNOWN_FIELD_TYPES.CONFLICT:
       return i18n.translate('discover.fieldNameIcons.conflictFieldAriaLabel', {
         defaultMessage: 'Conflicting field',
       });
-    case 'date':
+    case KNOWN_FIELD_TYPES.DATE:
       return i18n.translate('discover.fieldNameIcons.dateFieldAriaLabel', {
         defaultMessage: 'Date field',
       });
-    case 'geo_point':
+    case KNOWN_FIELD_TYPES.GEO_POINT:
       return i18n.translate('discover.fieldNameIcons.geoPointFieldAriaLabel', {
         defaultMessage: 'Geo point field',
       });
-    case 'geo_shape':
+    case KNOWN_FIELD_TYPES.GEO_SHAPE:
       return i18n.translate('discover.fieldNameIcons.geoShapeFieldAriaLabel', {
         defaultMessage: 'Geo shape field',
       });
-    case 'ip':
+    case KNOWN_FIELD_TYPES.IP:
       return i18n.translate('discover.fieldNameIcons.ipAddressFieldAriaLabel', {
         defaultMessage: 'IP address field',
       });
-    case 'murmur3':
+    case KNOWN_FIELD_TYPES.MURMUR3:
       return i18n.translate('discover.fieldNameIcons.murmur3FieldAriaLabel', {
         defaultMessage: 'Murmur3 field',
       });
-    case 'number':
+    case KNOWN_FIELD_TYPES.NUMBER:
       return i18n.translate('discover.fieldNameIcons.numberFieldAriaLabel', {
         defaultMessage: 'Number field',
       });
-    case 'source':
+    case KNOWN_FIELD_TYPES.SOURCE:
       // Note that this type is currently not provided, type for _source is undefined
       return i18n.translate('discover.fieldNameIcons.sourceFieldAriaLabel', {
         defaultMessage: 'Source field',
       });
-    case 'string':
+    case KNOWN_FIELD_TYPES.STRING:
       return i18n.translate('discover.fieldNameIcons.stringFieldAriaLabel', {
         defaultMessage: 'String field',
       });
-    case 'text':
+    case KNOWN_FIELD_TYPES.TEXT:
       return i18n.translate('discover.fieldNameIcons.textFieldAriaLabel', {
         defaultMessage: 'Text field',
       });
-    case 'keyword':
+    case KNOWN_FIELD_TYPES.KEYWORD:
       return i18n.translate('discover.fieldNameIcons.keywordFieldAriaLabel', {
         defaultMessage: 'Keyword field',
       });
-
-    case 'nested':
+    case KNOWN_FIELD_TYPES.NESTED:
       return i18n.translate('discover.fieldNameIcons.nestedFieldAriaLabel', {
         defaultMessage: 'Nested field',
       });
-    case 'version':
+    case KNOWN_FIELD_TYPES.VERSION:
       return i18n.translate('discover.fieldNameIcons.versionFieldAriaLabel', {
         defaultMessage: 'Version field',
       });
     default:
       return i18n.translate('discover.fieldNameIcons.unknownFieldAriaLabel', {
-        defaultMessage: 'Unknown field',
+        defaultMessage: UNKNOWN_FIELD_TYPE_MESSAGE,
```

Comment:
this is breaking the build. when you revert the change your tests would no longer work. So you could rewrite the test to first get the default message `getFieldTypeName()`, and check that all known fieldtypes return different values.

Reply:
Broken tests should now be fixed